### ISSUE TITLE
[Bug] setExtraLibs/addExtraLib does not update/remove existing extra libs

### ISSUE THREAD

**Issue by schoero** [microsoft/monaco-editor#2936]
### Reproducible in vscode.dev or in VS Code Desktop?

- [X] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [ ] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Monaco Editor Playground Code

```typescript
const editor = monaco.editor.create(document.getElementById("container"), {
    value: `test`,
    language: "typescript"
});

monaco.languages.typescript.typescriptDefaults.setExtraLibs([{ content: `declare const test = "initial";` }])

setTimeout(() => {
    monaco.languages.typescript.typescriptDefaults.setExtraLibs([{ content: `declare const test = "updated";` }])
}, 5000);
```

### Actual Behavior

When hovering over the variable `test` in the playground, the initial declaration `const test: "initial"` from the first setExtraLibs call gets applied as it should.
After a timeout of 5000ms setExtraLibs is called a second time and the declaration should change to `const test: "updated"` but it stays the same.

### Expected Behavior

The extra libs should be replaced with every subsequent setExtraLibs call but they actually remain on the values from the initial setExtraLibs call.

### Additional Context

The same thing seems to happen when using addExtraLib, which also has no effect on the extra libs.

**Comment by oldrich-s** [comment]
It was probably fixed in the latest update a couple of hours old.

**Comment by schoero** [comment]
Yes it is fixed in v0.32.0. Thank you.
